src/Host/Broker/Impl/Startup/Program.cs

```diff
@@ -57,7 +61,38 @@ public class Program {
 
             var tlsConfig = new TlsConfiguration(_logger, _securityOptions);
             var httpsOptions = tlsConfig.GetHttpsOptions(Configuration);
-            CreateWebHost(httpsOptions).Run();
+
+            Uri uri = GetServerUri(Configuration);
+            if (IsPortAvailable(uri?.Port)) {
```

Comment:
This is still fundamentally a race condition, right? Since any other process can grab the port between the check and the creation. Can we somehow handle the error that happens during the creation if the port ends up being taken, to mitigate that corner case?